// primeng-extensions/components/qrcode/qrcode.js
import { NgModule, Component, ElementRef, Input, ViewChild } from '@angular/core';
import { CommonModule } from '@angular/common';
var QRCode = (function () {
    function QRCode(el) {
        this.el = el;
        this.background = 'white';
        this.backgroundAlpha = 1.0;
        this.foreground = 'black';
        this.foregroundAlpha = 1.0;
        this.level = 'L';
        this.mime = 'image/png';
        this.padding = null;
        this.size = 100;
        this.initialized = false;
    }
    QRCode.prototype.ngOnInit = function () {
    };
    QRCode.prototype.ngAfterViewChecked = function () {
        if (!this.initialized) {
            this.init();
        }
    };
    QRCode.prototype.init = function () {
        this.initialized = true;
        var qrObject = {
            element: this.containerViewChild.nativeElement,
            background: this.background,
            backgroundAlpha: this.backgroundAlpha,
            foreground: this.foreground,
            foregroundAlpha: this.foregroundAlpha,
            level: this.level,
            mime: this.mime,
            padding: this.padding,
            size: this.size,
            value: this.value
        };
        this.qrcode = new QRious(qrObject);
    };
    QRCode.prototype.ngOnDestroy = function () {
        this.qrcode = null;
        this.initialized = false;
    };
    QRCode.decorators = [
        { type: Component, args: [{
                    selector: 'pe-qrcode',
                    template: "\n    <div class=\"ui-widget ui-corner-all\">\n      <canvas #qrcode></canvas>\n    </div>"
                },] },
    ];
    /** @nocollapse */
    QRCode.ctorParameters = function () { return [
        { type: ElementRef, },
    ]; };
    QRCode.propDecorators = {
        "value": [{ type: Input },],
        "background": [{ type: Input },],
        "backgroundAlpha": [{ type: Input },],
        "foreground": [{ type: Input },],
        "foregroundAlpha": [{ type: Input },],
        "level": [{ type: Input },],
        "mime": [{ type: Input },],
        "padding": [{ type: Input },],
        "size": [{ type: Input },],
        "containerViewChild": [{ type: ViewChild, args: ['qrcode',] },],
    };
    return QRCode;
}());
export { QRCode };
var QRCodeModule = (function () {
    function QRCodeModule() {
    }
    QRCodeModule.decorators = [
        { type: NgModule, args: [{
                    imports: [CommonModule],
                    exports: [QRCode],
                    declarations: [QRCode]
                },] },
    ];
    /** @nocollapse */
    QRCodeModule.ctorParameters = function () { return []; };
    return QRCodeModule;
}());
export { QRCodeModule };
