// primeng-extensions/components/badge/badge.js
import { NgModule, Component, ElementRef, Input, ViewEncapsulation } from '@angular/core';
import { CommonModule } from '@angular/common';
var Badge = (function () {
    function Badge(el) {
        this.el = el;
        this.backgroundColor = '';
    }
    Badge.prototype.data = function () {
        if (this.dot) {
            return;
        }
        if (typeof this.value === 'number' && typeof this.max === 'number') {
            return this.max < this.value ? this.max + "+" : this.value;
        }
        else {
            return this.value;
        }
    };
    Badge.prototype.display = function () {
        return (this.data() || this.dot);
    };
    Badge.prototype.ngOnDestroy = function () {
    };
    Badge.prototype.ngOnInit = function () {
    };
    Badge.decorators = [
        { type: Component, args: [{
                    selector: 'pe-badge',
                    template: "<div class=\"ui-badge ui-position\">\n        <ng-content></ng-content>\n        <span *ngIf=\"display()\"\n                [class]=\"badgeClass\" [ngStyle]=\"{'background-color': backgroundColor}\"\n                [ngClass]=\"{'ui-widget ui-widget-content ui-corner-all ui-badge-content':true,'fixed': true, 'dot': dot }\">\n            {{data()}}\n      </span>\n    </div>",
                    encapsulation: ViewEncapsulation.None,
                },] },
    ];
    /** @nocollapse */
    Badge.ctorParameters = function () { return [
        { type: ElementRef, },
    ]; };
    Badge.propDecorators = {
        "value": [{ type: Input },],
        "max": [{ type: Input },],
        "dot": [{ type: Input },],
        "backgroundColor": [{ type: Input },],
        "badgeClass": [{ type: Input },],
    };
    return Badge;
}());
export { Badge };
var BadgeModule = (function () {
    function BadgeModule() {
    }
    BadgeModule.decorators = [
        { type: NgModule, args: [{
                    imports: [CommonModule],
                    exports: [Badge],
                    declarations: [Badge]
                },] },
    ];
    /** @nocollapse */
    BadgeModule.ctorParameters = function () { return []; };
    return BadgeModule;
}());
export { BadgeModule };
